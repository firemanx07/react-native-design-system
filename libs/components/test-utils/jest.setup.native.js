import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = () => ({
  start: () => jest.fn(),
});
