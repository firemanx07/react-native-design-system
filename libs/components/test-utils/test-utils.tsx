import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
