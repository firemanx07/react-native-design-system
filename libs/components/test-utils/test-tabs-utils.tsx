import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
