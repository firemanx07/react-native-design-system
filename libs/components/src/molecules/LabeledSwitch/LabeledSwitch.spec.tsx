import { testAllStories } from '../../../test-utils';
import * as stories from './LabledSwitch.stories';

describe('Labeled Switch tests', () => {
  testAllStories(stories);
});
