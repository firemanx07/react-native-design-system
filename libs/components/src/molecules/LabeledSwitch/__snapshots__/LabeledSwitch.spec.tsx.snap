// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labeled Switch tests LabeledSwitch snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Rubik-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Labeled Switch:
  </Text>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#DEDFE0",
            "borderRadius": 7,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "minHeight": 40,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        fontFamily="Rubik-SemiBold"
        fontSize={12}
        isRight={true}
        lineHeight={18}
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-SemiBold",
              "fontSize": 12,
              "lineHeight": 18,
            },
            {
              "color": "rgba(44,44,44,0.6)",
            },
          ]
        }
      >
        Branches
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#4D37D9",
            "borderRadius": 8,
            "height": 16,
            "opacity": 1,
            "paddingBottom": 2,
            "paddingLeft": 2,
            "paddingRight": 2,
            "paddingTop": 2,
            "width": 34,
          }
        }
      >
        <View
          collapsable={false}
          size={12}
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 6,
              "elevation": 2,
              "height": 12,
              "shadowColor": "#2C2C2C",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "transform": [
                {
                  "translateX": 17,
                },
              ],
              "width": 12,
            }
          }
        />
      </View>
      <Text
        fontFamily="Rubik-SemiBold"
        fontSize={12}
        isRight={true}
        lineHeight={18}
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-SemiBold",
              "fontSize": 12,
              "lineHeight": 18,
            },
            {
              "color": "#2C2C2C",
            },
          ]
        }
      >
        ATM
      </Text>
    </View>
  </View>
</View>
`;
